lib/hashie/rash.rb

```diff
@@ -136,7 +136,7 @@ def respond_to_missing?(*args)
 
     def optimize_if_necessary!
       return unless (@lookups += 1) >= @optimize_every
-      @regexes = @regex_counts.sort_by { |_, count| -count }.map { |regex, _| regex }
+      @regexes = @regexes.sort_by { |regex| -@regex_counts[regex] }
```

Comment:
My initial thought is that we would want to call `#to_i` on the result of `@regex_counts[regex]` in case the two variables get desynchronized, but then I noticed that `@regex_counts` is initialized with `Hash.new(0)`, thus making sure it returns `0` instead of `nil`. Nice use of the default! 👍